dask_cloudprovider/providers/azure/setup/start_scheduler.py

```diff
@@ -0,0 +1,143 @@
+# Imports 
+import os
+import sys
+import uuid
+import time
+import socket
+import argparse
+import threading
+import subprocess
+
+from mpi4py import MPI
+from azureml.core import Run
+from notebook.notebookapp import list_running_servers
+
+def flush(proc, proc_log):
+    while True:
+        proc_out = proc.stdout.readline()
+        if proc_out == '' and proc.poll() is not None:
+            proc_log.close()
+            break
+        elif proc_out:
+            sys.stdout.write(proc_out)
+            proc_log.write(proc_out)
+            proc_log.flush()
+
+if __name__ == '__main__':
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank()
+
+    ### PARSE ARGUMENTS
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--jupyter",         default=False)
+    parser.add_argument("--code_store",      default=None)
+    parser.add_argument("--jupyter_token",   default=uuid.uuid1().hex)
+    parser.add_argument("--jupyter_port",    default=8888)
+    parser.add_argument("--dashboard_port",  default=8787)
+    parser.add_argument("--scheduler_port",  default=8786)
+    parser.add_argument("--use_gpu",         default=False)
```

Comment:
should remove the gpu arguments

Reply:
These arguments should stay here. We infer them in the azureml.py so these are transparent to the end user. Same happens for the number of GPUs to use in each worker.